React Docs: Update the "Columns" section. (#9752)
handsontable/handsontable: docs/content/guides/cell-functions/cell-renderer.md

type: 'numeric'
```
:::
:::

> 

Reviewer: ```suggestion
::: only-for react
## Overview

A renderer is a function that determines how a cell looks.

Set together, a renderer, [editor](@/guides/cell-functions/cell-editor.md) and [validator](@/guides/cell-functions/cell-validator.md) form a [cell type](@/guides/cell-types/cell-type.md).
:::
```

I know our changes are just a base for further content improvements, but we need some sort of introduction here.

::: only-for react

## Declaring a custom renderer as a component

> The React wrapper allows creating custom renderers using React components.

Reviewer: ```suggestion
Handsontable's React wrapper lets you create custom cell renderers using React components.
```

Although it's possible to use class-based react components for this purpose, **we strongly suggest** using functional components, as using the `state` of a class-based component would re-initialize on every Handsontable render.

To mark a component as a Handsontable renderer, simply add a `hot-renderer` attribute to it.

::: tip

> Handsontable's `autoRowSize` and `autoColumnSize` options require calculating the widths/heights of some of the cells before rendering them into the table. For this reason, it's not currently possible to use them alongside component-based renderers, as they're created after the table's initialization.

Reviewer: ```suggestion
Handsontable's [`autoRowSize`](@/api/options.md#autorowsize] and [`autoColumnSize`](@/api/options.md#autocolumnsize) options require calculating the widths/heights of some of the cells before rendering them into the table. For this reason, it's not currently possible to use them alongside component-based renderers, as they're created after the table's initialization.
```

> Be sure to turn those options off in your Handsontable config, as keeping them enabled may cause unexpected results. Please note that `autoColumnSize` is enabled by default.

Reviewer: ```suggestion
Be sure to turn those options off in your Handsontable configuration, as keeping them enabled may cause unexpected results. Please note that [`autoColumnSize`](@/api/options.md#autocolumnsize) is enabled by default.
```

import { HotTable, HotColumn } from '@handsontable/react';
import 'handsontable/dist/handsontable.min.css';

// your renderer component
const RendererComponent = (props) => {
  // the available renderer-related props are:
  // - `row` (row index)
  // - `col` (column index)
  // - `prop` (column property name)
  // - `TD` (the HTML cell element)
  // - `cellProperties` (the `cellProperties` object for the edited cell)
  return (
    <React.Fragment>
      <i style={{ color: "#a9a9a9" }}>
        Row: {props.row}, column: {props.col},
      </i>{" "}
      value: {props.value}
    </React.Fragment>
  );
}

const hotData = Handsontable.helper.createSpreadsheetData(10, 5);

const ExampleComponent = () => {
  return (
    <div>
    <HotTable data={hotData} licenseKey="non-commercial-and-evaluation">
      <HotColumn width={250}>
        {/* add the `hot-renderer` attribute to mark the component as a Handsontable renderer */}
        <RendererComponent hot-renderer />
      </HotColumn>
    </HotTable>
    </div>
  );
};

ReactDOM.render(<ExampleComponent />, document.getElementById('example1'));
```
:::

## Using the renderer component within React's Context

> In this example, React's Context is used to pass the information available in the main app component to the renderer. In this case, we're using just the renderer, but the same principle works with editors just as well.

Reviewer: ```suggestion
In this example, React's `Context` passes information available in the main app component to the renderer. In this case, we're using just the renderer, but the same principle works with [editors](@/guides/cell-functions/cell-editor.md) as well.
```
